zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -161,12 +161,28 @@ class DataStoreManager(metaDataset: String,
     val dropTableName = dropTable.dataset
 
     if(metaData.contains(dropTableName)){
-      metaData.remove(dropTableName)
+      val metaRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dropTableName))
+      metaActor ! DeleteRecord(metaDataset, Seq(metaRecordFilter))
+
+      metaData.foreach( dataSchema =>
+        dataSchema._2.createQueryOpt match {
+          case Some(query) =>
+            if(query.dataset == dropTableName){
+              metaActor ! DropView(dataSchema._1)
+              val viewRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dataSchema._1))
+              metaActor ! DeleteRecord(metaDataset, Seq(viewRecordFilter))
```

Comment:
It's created [here](https://github.com/ISG-ICS/cloudberry/blob/master/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala#L183). Yes, one agent is responsible for one specific dataset. You can just get the actor by `context.child("data-" + query.dataset)`. and send it a `PoisonKill` msg.